From 55bdaaf8368dec98c13b722b938571a6aed4ba55 Mon Sep 17 00:00:00 2001
From: Marius Vlad <marius.vlad@collabora.com>
Date: Fri, 5 Mar 2021 22:07:30 +0200
Subject: [PATCH 8/8] libweston: Rename weston_view_activate() to
 weston_view_activate_input()

This way, we try to differentiate between input focus and window/surface
activation.

Upstream-Status: Pending MR [https://gitlab.freedesktop.org/wayland/weston/-/merge_requests/570]
Signed-off-by: Marius Vlad <marius.vlad@collabora.com>
---
 desktop-shell/shell.c         | 2 +-
 include/libweston/libweston.h | 6 +++---
 kiosk-shell/kiosk-shell.c     | 6 +++---
 libweston/input.c             | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index 1f11e088..15379e1f 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -3840,7 +3840,7 @@ activate(struct desktop_shell *shell, struct weston_view *view,
 	if (shsurf->output)
 		lower_fullscreen_layer(shell, shsurf->output);
 
-	weston_view_activate(view, seat, flags);
+	weston_view_activate_input(view, seat, flags);
 
 	if (shseat->focused_surface) {
 		struct shell_surface *current_focus =
diff --git a/include/libweston/libweston.h b/include/libweston/libweston.h
index dc54b170..066d8b35 100644
--- a/include/libweston/libweston.h
+++ b/include/libweston/libweston.h
@@ -1561,9 +1561,9 @@ weston_view_from_global_fixed(struct weston_view *view,
 			      wl_fixed_t *vx, wl_fixed_t *vy);
 
 void
-weston_view_activate(struct weston_view *view,
-		     struct weston_seat *seat,
-		     uint32_t flags);
+weston_view_activate_input(struct weston_view *view,
+		           struct weston_seat *seat,
+		           uint32_t flags);
 
 void
 notify_modifiers(struct weston_seat *seat, uint32_t serial);
diff --git a/kiosk-shell/kiosk-shell.c b/kiosk-shell/kiosk-shell.c
index c73c6268..981cd5a7 100644
--- a/kiosk-shell/kiosk-shell.c
+++ b/kiosk-shell/kiosk-shell.c
@@ -615,7 +615,7 @@ desktop_surface_removed(struct weston_desktop_surface *desktop_surface,
 		wl_list_for_each(seat, &shell->compositor->seat_list, link) {
 			struct weston_keyboard *keyboard = seat->keyboard_state;
 			if (keyboard && keyboard->focus == surface)
-				weston_view_activate(focus_view, seat, 0);
+				weston_view_activate_input(focus_view, seat, 0);
 		}
 	}
 
@@ -671,7 +671,7 @@ desktop_surface_committed(struct weston_desktop_surface *desktop_surface,
 
 		wl_list_for_each(seat, &shsurf->shell->compositor->seat_list, link) {
 			struct kiosk_shell_seat *kiosk_seat = get_kiosk_shell_seat(seat);
-			weston_view_activate(shsurf->view, seat, 0);
+			weston_view_activate_input(shsurf->view, seat, 0);
 
 			if (kiosk_seat->focused_surface) {
 				struct kiosk_shell_surface *current_focus =
@@ -896,7 +896,7 @@ kiosk_shell_activate_view(struct kiosk_shell *shell,
 		weston_surface_damage(view->surface);
 	}
 
-	weston_view_activate(view, seat, flags);
+	weston_view_activate_input(view, seat, flags);
 }
 
 static void
diff --git a/libweston/input.c b/libweston/input.c
index 28dcb0b9..b340a5ea 100644
--- a/libweston/input.c
+++ b/libweston/input.c
@@ -1861,7 +1861,7 @@ inc_activate_serial(struct weston_compositor *c)
 }
 
 WL_EXPORT void
-weston_view_activate(struct weston_view *view,
+weston_view_activate_input(struct weston_view *view,
 		     struct weston_seat *seat,
 		     uint32_t flags)
 {
-- 
2.30.1

